pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -398,6 +402,11 @@ type HAProxySpec struct {
 	ReplicasServiceEnabled *bool `json:"replicasServiceEnabled,omitempty"`
 }
 
+type ProxySQLSpec struct {
+	PodSpec   `json:",inline"`
+	Scheduler string `json:"scheduler,omitempty"`
```

Comment:
let's leverage k8s API validation 
```suggestion
        // +kubebuilder:validation:Enum={percona,proxysql-admin}
        Scheduler string `json:"scheduler,omitempty"`
```

this way users will see an error like:
```
The PerconaXtraDBCluster "cluster1" is invalid: spec.proxysql.scheduler: Unsupported value: "ege": supported values: "percona", "proxysql-admin"
```

we can remove our own validation

Reply:
Added in: 048c372ce5273e1c826543e3b832b7474f7bc08a